Getting started doc: ubuntu dependencies and formatting (#5672)
sourcegraph/sourcegraph: doc/dev/local_development.md

# Getting started with developing Sourcegraph

> Have a look around, our code is at https://sourcegraph.com/github.com/sourcegraph/sourcegraph.

Reviewer: ```suggestion
Have a look around, our code is on [GitHub](https://sourcegraph.com/github.com/sourcegraph/sourcegraph).
```

- [nginx](https://docs.nginx.com/nginx/admin-guide/installing-nginx/installing-nginx-open-source/) (v1.14 or higher)
- [SQLite](https://www.sqlite.org/index.html) tools

> The following are two recommended options for installing these dependencies:

Reviewer: ```suggestion
The following are two recommendations for installing these dependencies:
```

    # Yarn
    curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | sudo apt-key add -
    echo "deb https://dl.yarnpkg.com/debian/ stable main" | sudo tee /etc/apt/sources.list.d/yarn.list

    # Node.js
    curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
    ```

2. Update repositories:

    ```bash
    sudo apt-get update
    ```

3. Install dependencies:

    ```bash
    sudo apt install -y make git-all postgresql postgresql-contrib redis-server nginx libpcre3-dev libsqlite3-dev pkg-config golang-go musl-tools docker-ce docker-ce-cli containerd.io nodejs yarn
    ```

4. Configure startup services

    ```bash
    sudo systemctl enable postgresql
    sudo systemctl enable redis-server.service
    ```

5. (optional) You can also run Redis using Docker

    In this case you should not enable the `redis-server.service` from the previous step.

    ```bash
    dockerd # if docker isn't already running
    docker run -p 6379:6379 -v $REDIS_DATA_DIR redis
    # $REDIS_DATA_DIR should be an absolute path to a folder where you intend to store Redis data
    ```

>     You need to have Redis running when you start the dev server later on. If you have issues running Docker, try [adding your user to the docker group][dockerGroup], and/or [updating the socket file persimissions][socketPermissions], or try running these commands under `sudo`.

Reviewer: ```suggestion
You need to have Redis running when you start the dev server later on. If you have issues running Docker, try [adding your user to the docker group][dockerGroup], and/or [updating the socket file persimissions][socketPermissions], or try running these commands under `sudo`.
```

>     You need to have Redis running when you start the dev server later on. If you have issues running Docker, try [adding your user to the docker group][dockerGroup], and/or [updating the socket file persimissions][socketPermissions], or try running these commands under `sudo`.

Reviewer: Is this meant to be url?

>     [dockerGroup]: https://stackoverflow.com/a/48957722

Reviewer: Could you add these links to the above paragraph so this can be removed?
